File: pq.cs
public static double[] pq(double p, double q)
{
  if(Math.Pow(p/2, 2) - q >= 0)
    return new double[] {
      -p/2 - Math.Sqrt(Math.Pow(p/2, 2) - q ),
      -p/2 + Math.Sqrt(Math.Pow(p/2, 2) - q )
    };
   else throw new Exception("Keine Nullstelle!");
}

File: pq.fs
open System

let pq(p : double, q : double) = 
    if (Math.Pow(p/2.0, 2.0) - q >= 0.0) then
        [|
            -p/2.0 - Math.Sqrt(Math.Pow(p/2.0, 2.0) - q);
            -p/2.0 + Math.Sqrt(Math.Pow(p/2.0, 2.0) - q)
        |]
    else
        raise(new Exception("Keine Nullstelle"))
